src/Formats/NativeWriter.cpp

```diff
@@ -70,6 +70,7 @@ static void writeData(const ISerialization & serialization, const ColumnPtr & co
     settings.low_cardinality_max_dictionary_size = 0;
     settings.data_types_binary_encoding = format_settings && format_settings->native.encode_types_in_binary_format;
     settings.write_json_as_string = format_settings && format_settings->native.write_json_as_string;
+    settings.use_v1_object_and_dynamic_serialization = client_revision < DBMS_MIN_REVISION_WITH_V2_DYNAMIC_AND_JSON_SERIALIZATION;
```

Comment:
so it's rather old client - new server fix, not new client - old server (like PR description says)?

Reply:
No, it's actually when new client sends data to old server using Native protocol. In this case `client_revision` here will be the revision of the server we send the data to.